js/main.js

```diff
@@ -420,6 +428,38 @@ function renderPokemonDetails(responseData) {
 
   $pokemonDetail.appendChild($pokemonDetailsBorder);
 
+  var $star = document.getElementById('star');
+
+  var starClicked = false;
+  for (var p = 0; p < data.favorited.length; p++) {
+    if (data.currentPokemon.id === data.favorited[p].id) {
+      $star.setAttribute('class', 'fa-solid fa-star star-color-goldenrod star-size padding-top-right-very-small margin-star');
+      starClicked = true;
+    }
+  }
+  function starTest(event) {
+    if (starClicked === false) {
+      $star.setAttribute('class', 'fa-solid fa-star star-color-goldenrod star-size padding-top-right-very-small margin-star');
+      var favoritedPokemon = {};
+      var splitHeading = $h1.textContent.split(' ');
+      favoritedPokemon.id = splitHeading[0];
+      favoritedPokemon.name = splitHeading[1];
+      data.favorited.unshift(favoritedPokemon);
+      starClicked = true;
+    } else {
+      $star.setAttribute('class', 'fa-solid fa-star star-color-grey star-size padding-top-right-very-small margin-star');
+      for (var i = 0; i < data.favorited.length; i++) {
+        if (data.favorited[i].id === data.currentPokemon.id) {
+          data.favorited.splice(i, 1);
+        }
+      }
+      starClicked = false;
+    }
+
+  }
```

Comment:
You generally should not have functions inside of other functions. If you are doing this to have access to `starClicked`, you should make it part of your data model instead. 

However, I could see using a single variable to check if a star was clicked or not causing issues as you move between different pokemon. Instead, I would search your list of favorites for the pokemon you are showing details on when you go to render the DOM tree, and if it is already in favorites, make the star filled, and if it isn't, make it hollow.